server/src/test/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdaterTests.java

```diff
@@ -471,4 +449,108 @@ public void testRemovingArchivedSettingsDoesNotRemoveNonArchivedInvalidOrUnknown
         }
     }
 
+    private static List<Setting<String>> createUnknownSettings(int numberOfUnknownSettings) {
+        final List<Setting<String>> unknownSettings = new ArrayList<>(numberOfUnknownSettings);
+        for (int i = 0; i < numberOfUnknownSettings; i++) {
+            final Setting<String> unknownSetting = Setting.simpleString("unknown.setting" + i, Property.NodeScope);
+            unknownSettings.add(unknownSetting);
+        }
+        return unknownSettings;
+    }
+
+    private static List<Setting<String>> createInvalidSettings(int numberOfInvalidSettings) {
+        final List<Setting<String>> invalidSettings = new ArrayList<>(numberOfInvalidSettings);
+        for (int i = 0; i < numberOfInvalidSettings; i++) {
+            final Setting<String> invalidSetting = createInvalidSetting(i);
+            invalidSettings.add(invalidSetting);
+        }
+        return invalidSettings;
+    }
+
+    private static Setting<String> createInvalidSetting(int index) {
```

Comment:
Could we distinguish invalid-in-isolation settings from invalid-with-dependencies settings to make sure we're doing both kinds of validation as appropriate?

Reply:
Yes, it will be better :+1:

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/34184/commits/2efdf1e79f66e43f5b9a1450542b85f018d6a6ae#diff-7cc27247465d367dd79b97b65722eae1R463 with random creation pattern.